src/MongoDB.Driver.Legacy/Builders/IndexKeysBuilder.cs

```diff
@@ -41,6 +41,16 @@ public static IndexKeysBuilder Ascending(params string[] names)
             return new IndexKeysBuilder().Ascending(names);
         }
 
+        /// <summary>
+        /// Sets wildcard keys to index in ascending order. If the wildcard names collection is empty, the generated key will be for the whole collection.
+        /// </summary>
+        /// <param name="names">The wildcard keys.</param>
+        /// <returns>The builder (so method calls can be chained).</returns>
+        public static IndexKeysBuilder AscendingWithWildcard(params string[] names)
```

Comment:
I'm not sure that this is the best name for this method. So, I will be happy to listen to other suggestions.

Reply:
Also, the logic for `Ascending` method seems strange for me since in case if the `names` collection is empty, we won't do anything implicitly.

Reply:
I agree that if you call `Ascending` with a zero length array the results are undefined. But maybe that's not relevant to CSHARP-2408.